cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py

```diff
@@ -0,0 +1,118 @@
+                #           * creation of slurmdbd.conf;
+                #           * start of slurmdbd;
+                #           * minimal bootstrapping of Slurm Accounting database.
+                #        * configuration of CloudWatch log group creation and slurmdbd log push to the CW log group (this require appropriate IAM policies).
+            },
+            "configure": {
+                "files": {
+                    "/etc/cfn/hooks.d/cfn-auto-reloader.conf": {
+                        "content": Fn.sub(
+                            "[cfn-auto-reloader-hook]\n"
+                            "triggers=post.update\n"
+                            "path=Resources.LaunchTemplate.Metadata.AWS::CloudFormation::Init\n"
+                            "action=/usr/bin/echo 'I was triggered by a change in AWS::CloudFormation::Init metadata!' > /tmp/cfn_init_metadata_update.log\n"
+                            "runas=root\n"
+                        ),
+                        "mode": "000400",
+                        "owner": "root",
+                        "group": "root",
+                    }
+                }
+            },
+        }
+
+    def _add_external_slurmdbd_target_group(self):
+        return elbv2.NetworkTargetGroup(
+            self,
+            "External-Slurmdbd-TG",
+            health_check=elbv2.HealthCheck(
+                port="22",
+                protocol=elbv2.Protocol.TCP,
+            ),
+            port=22,
+            protocol=elbv2.Protocol.TCP,
+            target_type=elbv2.TargetType.INSTANCE,
+            vpc=self.vpc,
+        )
+
+    def _add_external_slurmdbd_launch_template(self):
+        # Define a CfnParameter for the AMI ID
+        # This AMI should be Parallel Cluster AMI, which has installed Slurm and related software
+        ami_id_param = CfnParameter(self, "AmiId", type="String", description="The AMI id for the EC2 instance.")
+
+        launch_template_data = ec2.CfnLaunchTemplate.LaunchTemplateDataProperty(
+            image_id=ami_id_param.value_as_string,
+        )
+
+        launch_template = ec2.CfnLaunchTemplate(
+            self,
+            "LaunchTemplate",
+            launch_template_data=launch_template_data,
+        )
+
+        launch_template.add_metadata("AWS::CloudFormation::Init", self._cfn_init_config)
+
+        return launch_template
+
+    def _add_external_slurmdbd_load_balancer(
+        self,
+        target_group,
+    ):
+        nlb = elbv2.NetworkLoadBalancer(self, "External-Slurmdbd-NLB", vpc=self.vpc, internet_facing=False)
+
+        # add listener to NLB
+        listener = nlb.add_listener("External-Slurmdbd-Listener", port=22)
+        listener.add_target_groups("External-Slurmdbd-Target", target_group)
+
+        return nlb
+
+    def _add_external_slurmdbd_auto_scaling_group(self):
+        return autoscaling.CfnAutoScalingGroup(
+            self,
+            "External-Slurmdbd-ASG",
+            max_size="1",
```

Comment:
We should add a comment of why we want the ASG to have 1 as the min, max and desired capacity so that there is always one SlurmDBD instance

Reply:
The commit is outdated